### PR TITLE
Re-lower the MSRV to 1.48

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

**Comment by alex** [comment]
Clippy failures are addressed by https://github.com/sfackler/rust-openssl/pull/1807

**Comment by alex** [comment]
The no-op bssl-sys crate from crates.io has a 2021 edition, I've reached out to the maintainer to see about lowering that.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,11 +57,16 @@ jobs:
   min-version:
     name: min-version
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # Current MSRV and expected next MSRV
+        RUST: ["1.48.0", "1.56.0"]
```
Comment (sfackler) [inline .github/workflows/ci.yml:64]: Is 1.56 actually an expected next MSRV? that was IIRC just the one that one of those dependency minor versions required. I don't think it was any more intentional than that.

Comment (alex) [reply .github/workflows/ci.yml:64]: It's pyca/cryptography's next expected MSRV (based on our usage numbers). So we'd be happy with that reality :-)

Comment (sfackler) [reply .github/workflows/ci.yml:64]: Fair enough!

**Pull request closed without merging**